### PR TITLE
fix(filter): add the "filled" class for styling purposes 

### PR DESCRIPTION
**Pull request by mcallegario**
Fix bug when using "updateFilters" method on fields with selection filter in adding "filled" class for styling purposes

### CODE REVIEW

File: packages/common/src/filters/selectFilter.ts [hunk 1]
```diff
@@ -229,8 +229,14 @@ export class SelectFilter implements Filter {
       values = Array.isArray(values) ? values : [values];
       this.$filterElm.multipleSelect('setSelects', values);
     }
-    this.getValues().length > 0 ? this.$filterElm.addClass('filled') : this.$filterElm.removeClass('filled');
-
+    if (this.getValues().length > 0 ) {
```
Comment (ghiscoding) [inline packages/common/src/filters/selectFilter.ts:232]: ESLint throws an error on this line `error  There should be no space before this paren  space-in-parens` and is caused by the space after `> 0 `, just remove the space and it will then run the full CI/CD and unit tests

Comment (mcallegario) [reply packages/common/src/filters/selectFilter.ts]: Oops!!! again... ;-)
Now that I have enabled the option in the VSCode of "Format on save", it will be much better.
Thanks

**Comment by ghiscoding** [comment]
We're missing 3 lines of tests, I can look at that later, so I won't merge right away since I want to keep my 100% coverage 😉

File: packages/common/src/filters/selectFilter.ts [hunk 2]
```diff
@@ -398,7 +404,7 @@ export class SelectFilter implements Filter {
 
     // if there's a search term, we will add the "filled" class for styling purposes
     if (this.isFilled) {
```
Comment (ghiscoding) [inline packages/common/src/filters/selectFilter.ts:406]: replace line 406 to 408 by 1 line
```diff
- if (this.isFilled) {
-    this.$filterElm?.addClass('filled').siblings('div .search-filter').addClass('filled');
- }
```
with this 1 line
```ts
this.updateFilterStyle(this.isFilled);
```
further below for the third and last part, replace lines 457 to 475
```diff
-  protected onTriggerEvent() {
-    if (this.$filterElm) {
-      const selectedItems = this.getValues();
-
-      if (Array.isArray(selectedItems) && selectedItems.length > 1 || (selectedItems.length === 1 && selectedItems[0] !== '')) {
-        this.isFilled = true;
-        this.$filterElm?.addClass('filled').siblings('div .search-filter').addClass('filled');
-      } else {
-        this.isFilled = false;
-        this.$filterElm.removeClass('filled');
-        this.$filterElm.siblings('div .search-filter').removeClass('filled');
-      }
-
-      this.searchTerms = selectedItems;
-      this.callback(undefined, { columnDef: this.columnDef, operator: this.operator, searchTerms: selectedItems, shouldTriggerQuery: this._shouldTriggerQuery });
-      // reset flag for next use
-      this._shouldTriggerQuery = true;
-    }
-  }
```
with this updated method and another new one
```ts
protected onTriggerEvent() {
    if (this.$filterElm) {
      const selectedItems = this.getValues();
      this.updateFilterStyle(Array.isArray(selectedItems) && selectedItems.length > 1 || (selectedItems.length === 1 && selectedItems[0] !== ''));
      this.searchTerms = selectedItems;
      this.callback(undefined, { columnDef: this.columnDef, operator: this.operator, searchTerms: selectedItems, shouldTriggerQuery: this._shouldTriggerQuery });
      // reset flag for next use
      this._shouldTriggerQuery = true;
    }
}

/** Set value(s) on the DOM element */
protected updateFilterStyle(isFilled: boolean) {
    if (isFilled) {
      this.isFilled = true;
      this.$filterElm?.addClass('filled').siblings('div .search-filter').addClass('filled');
    } else {
      this.isFilled = false;
      this.$filterElm.removeClass('filled');
      this.$filterElm.siblings('div .search-filter').removeClass('filled');
    }
}
```

File: packages/common/src/filters/selectFilter.ts [hunk 3]
```diff
@@ -229,8 +229,14 @@ export class SelectFilter implements Filter {
       values = Array.isArray(values) ? values : [values];
       this.$filterElm.multipleSelect('setSelects', values);
     }
-    this.getValues().length > 0 ? this.$filterElm.addClass('filled') : this.$filterElm.removeClass('filled');
```
Comment (ghiscoding) [inline packages/common/src/filters/selectFilter.ts:232]: @mcallegario I took a look at this and we're missing unit tests and I also see duplicate code for the same `filled` styling css class, so would it be possible to ask you to modify the code in 3 areas? If possible, that'd be great and the change could do under your name as contribution. 

The step 1 would be to delete line 232 to 239
```diff
-    if (this.getValues().length > 0) {
-      this.isFilled = true;
-      this.$filterElm?.addClass('filled').siblings('div .search-filter').addClass('filled');
-    } else {
-      this.isFilled = false;
-      this.$filterElm.removeClass('filled');
-      this.$filterElm.siblings('div .search-filter').removeClass('filled');
-    }
```
and replace with this, I'll show the other 2 steps on the next line number
```ts
this.updateFilterStyle(this.getValues().length > 0);
```

**Review by ghiscoding — CHANGES_REQUESTED** [review]

**Comment by mcallegario** [comment]
Ready!
I made the requested adjustments.

**Review by ghiscoding — APPROVED** [review]

**Comment by ghiscoding** [comment]
Awesome thanks, this change would probably have to be applied to all other Filters. Would you be interested in doing another PR? I can also do it myself, I'm just asking if you're interested to contribute even more :)

**Comment by mcallegario** [comment]
Thanks @ghiscoding !
Yes, it's a pleasure to help you. I will review all other filters.
